database/factories/ModelFactory.php

```diff
@@ -21,3 +26,72 @@
         'visit_token' => $faker->asciify('********************'),
     ];
 });
+
+$factory->define(App\DuesPackage::class, function (Faker\Generator $faker) {
+    return [
+        'id' => $faker->numberBetween($min = 0, $max = 200),
+        'name' => $faker->word,
+        'eligible_for_shirt' => $faker->numberBetween($min = 0, $max = 1),
+        'eligible_for_polo' => $faker->numberBetween($min = 0, $max = 1),
+        'effective_start' => $faker->dateTime(),
+        'effective_end' => $faker->dateTime(),
+        'cost' => (string) $faker->randomFloat($nbMaxDecimals = 2, $min = 0, $max = 1000),
+    ];
+});
+
+$factory->define(App\DuesTransaction::class, function (Faker\Generator $faker) {
+    return [
+        'id' => $faker->numberBetween($min = 0, $max = 200),
+        'swag_shirt_provided' => $faker->optional()->dateTime,
+        'swag_shirt_providedBy' => User::first() != null ? User::first()->id : null,
+        'swag_polo_provided' => $faker->optional()->dateTime,
+        'swag_polo_providedBy' => User::first() != null ? User::first()->id : null,
+        'dues_package_id' => $faker->randomElement($array = idMap(DuesPackage::all())),
+        'user_id' => $faker->randomElement($array = idMap(User::all())),
+    ];
+});
+
+$factory->define(App\User::class, function (Faker\Generator $faker) {
+    $lastName = $faker->lastName;
+    $uid = $faker->bothify('?'.lcfirst($lastName).'##');
+
+    return [
+        'id' => $faker->numberBetween($min = 0, $max = 200),
+        'uid' => $uid,
+        'gtid' => $faker->numerify('#########'),
+        'slack_id' => null,
+        'gt_email' => $uid.'@gatech.edu',
+        'personal_email' => $faker->safeEmail,
+        'first_name' => $faker->firstName,
+        'middle_name' => $faker->optional()->lastName,
+        'last_name' => $lastName,
+        'preferred_name' => $faker->firstName,
+        'phone' => $faker->numerify('##########'),
+        'emergency_contact_name' => null,
+        'emergency_contact_phone' => null,
+        'join_semester' => null,
+        'graduation_semester' => null,
+        'shirt_size' => $faker->randomElement($array = ['s', 'm', 'l', 'xl']),
+        'polo_size' => $faker->randomElement($array = ['s', 'm', 'l', 'xl']),
+        'gender' => $faker->randomElement($array = ['male', 'female', 'nonbinary', 'none']),
+        'ethnicity' => $faker->randomElement($array = ['white', 'asian', 'hispanic', 'black', 'native', 'islander', 'none']),
+        'accept_safety_agreement' => $faker->optional()->dateTime,
+    ];
+});
+
+$factory->define(App\Payment::class, function (Faker\Generator $faker) {
+    return [
+        'id' => $faker->numberBetween($min = 0, $max = 200),
+        'payable_id' => $faker->numberBetween($min = 0, $max = 200),
+        'payable_type' => $faker->randomElement($array = ['App\DuesTransaction', 'App\Event']),
```

Comment:
If the `DuesTransaction` or `Event` doesn't exist, this may not be of much use. Is that accounted for when this is used? I've not done much with model factories before.

Reply:
I believe I was doing this because an App\Payment requires a DuesTransaction or Event to exist to even be created properly, and I couldn't find a good way to autogenerate a DuesTransaction when generating a Payment. Not sure what to do given that dependency...

Reply:
Resolved by creating a `DatabaseSeeder`